src/PlatformEngines.jl

```diff
@@ -598,7 +598,7 @@ function get_server_dir(url::AbstractString, server=pkg_server())
         @warn "malformed Pkg server value" server
         return
     end
-    joinpath(depots1(), "servers", m.captures[1]::AbstractString)
+    joinpath(depots1(), "servers", String(m.captures[1]))
```

Comment:
This change probably isn't necessary:
```julia
julia> fieldtype(RegexMatch, :captures)
Vector{Union{Nothing, SubString{String}}} (alias for Array{Union{Nothing, SubString{String}}, 1})
```
and so the `::AbstractString` should be enough for Julia to know that it's a `SubString{String}`. If that's not true, then the best might be `m::RegexMatch`?

Reply:
The issue is that if it becomes a `SubString` then `joinpath` will not specialize on all arguments in `joinpath(::String, ::String, ::SubString{String}` but instead consider the last argument as a `Vararg{AbstractString, String}`.

Reply:
Got it, carry on!